src/Razor/src/Microsoft.VisualStudio.DevKit.Razor/Microsoft.VisualStudio.DevKit.Razor.csproj

```diff
@@ -52,11 +52,11 @@
 
       <!-- Transform RuntimeIdentifierForPublish items to project items to pass to MSBuild task -->
       <ProjectToPublish Include="@(RuntimeIdentifierForPublish->'$(MSBuildProjectFullPath)')">
-        <AdditionalProperties>RuntimeIdentifier=%(RuntimeIdentifierForPublish.Identity);PublishDir=$(RidsPublishDir)%(RuntimeIdentifierForPublish.Identity)\;TargetFramework=$(PublishTargetFramework);SelfContained=false</AdditionalProperties>
+        <AdditionalProperties>_IsPublishing=true;NoBuild=true;PublishRuntimeIdentifier=%(RuntimeIdentifierForPublish.Identity);PublishDir=$(RidsPublishDir)%(RuntimeIdentifierForPublish.Identity)\;TargetFramework=$(PublishTargetFramework);SelfContained=false;AppendRuntimeIdentifierToOutputPath=false</AdditionalProperties>
```

Comment:
is `AppendRuntimeIdentifierToOutputPath=false` the default, or is this a change to what we produce?

Reply:
This is a change to allow each RID-specific publish to reuse the (ridless) build from the SolutionBuild phase of the build.

Reply:
Those are words that I understand in isolation, but have no idea what they mean together.

Reply:
Before this PR, the build works as follows:

- When Arcade's Build.proj is built, Arcade builds the root solution (or the solution filter).
- On Windows, the solution includes both of the csproj projects, so both are built without a RID (so they are built into the normal output path without a RID)
- On Linux, the same thing happens for the DevKit project, but not rzls (as it's not in the solution filter)
- After Build.proj finishes building, then the AfterSolutionBuild.proj project is built
- AfterSolutionBuild.proj calls the `PublishAllRids` targets for DevKit and rzls.
- For both DevKit and rzls, the following happens:
  - For each RID:
    - Build the project and all dependencies again with each RID and append the RID to the intermediate and final output paths.
    - Publish this RID-based build from the output folder that has the RID appended.

After this PR, the build works as follows:

- When Arcade's Build.proj is built, Arcade builds the root solution (or the solution filter).
  - Build.proj builds both DevKit and rzls without a RID (so they are built into the normal output path without a RID)
- After Build.proj finishes building, then the AfterSolutionBuild.proj project is built
- AfterSolutionBuild.proj calls the `PublishAllRids` targets for DevKit and rzls.
- For both DevKit and rzls, the following happens:
  - For each RID:
    - Skip the build (because we specified `NoBuild=true`
    - Publish this RID-based build from the output folder that does not have the RID appended (the build we built earlier)

As a result, this ensures that we don't build these projects and all dependencies N (or N+1 on Windows) times for N rids. With this change, we'll build these projects 1 time and publish N times.